client/network/sync/src/lib.rs

```diff
@@ -1356,31 +1291,19 @@ where
 	}
 
 	fn send_block_request(&mut self, who: PeerId, request: BlockRequest<B>) {
-		let (tx, rx) = oneshot::channel();
-		let opaque_req = self.create_opaque_block_request(&request);
-
 		if self.peers.contains_key(&who) {
```

Comment:
I'm not entirely sure why the code has been written this way but `SyncingEngine` shouldn't be sending requests to peers it doesn't know about. I think what previously happened if the peer was not known, the request was dropped as the RX half is dropped. I don't imagine this will cause issues